### PR TITLE
utest: Revamp test framework

### PR DESCRIPTION
**Pull request by Skallwar**
(Don't look at the branch name)

### CODE REVIEW

**Comment by CohenArthur** [comment]
The use of these tests was to ensure that unit testing is operational and shows up. We can remove them if you feel like they're not important anymore, but they were simply here to ensure that no regression ever happens to the testing framework or even to tests themselves, considering it was something to even get the tests to run :D

**Comment by Skallwar** [comment]
> The use of these tests was to ensure that unit testing is operational and shows up. We can remove them if you feel like they're not important anymore, but they were simply here to ensure that no regression ever happens to the testing framework or even to tests themselves, considering it was something to even get the tests to run

I revamped the test framework and keep one test for self checking purpose. Are you ok with that ?

File: src/utest/mod.rs [hunk 1]
```diff
@@ -63,9 +51,6 @@ fn panic(info: &PanicInfo) -> ! {
 }
 
 #[test_case]
-fn utests_test() {
-    uassert_eq(true, true, "utest framework initialization");
-    kassert_eq!(true, true);
-    kassert!(true);
-    kassert_eq!(true, true, "utest framework initialization complete");
+fn test_slef_check() {
```
Comment (Skallwar) [inline src/utest/mod.rs:54]:
```suggestion
fn test_self_check() {
```

File: src/utest/mod.rs [hunk 2]
```diff
@@ -7,47 +7,35 @@ use crate::println;
...
-/// Assert the equality of two elements
-macro_rules! kassert_eq {
-    ($l_exp: expr, $r_exp: expr) => {{
-        crate::utest::uassert_eq($l_exp, $r_exp, "anonymous test")
-    }};
-    ($l_exp: expr, $r_exp: expr, $name: tt) => {{
-        crate::utest::uassert_eq($l_exp, $r_exp, $name)
-    }};
+pub trait Testable {
+    fn run(&self) -> ();
```
Comment (CohenArthur) [inline src/utest/mod.rs:14]:
```suggestion
    fn run(&self);
```

Any reason why this doesn't work or is different ?

**Review by CohenArthur — APPROVED** [review]
This is definitely black magic but it looks good!

**Pull request merged**